docs: add documentation for lazy load and dev workflow

williamareynolds opened the pull request.

william-reynolds-flosports committed:
docs: add development workflow to main project readme

patrickmichalina commented on README.md:
````diff
@@ -46,8 +47,116 @@
+add a new set of components to the app.
+
+```shell script
+ng g c your-library-name --module app
````
`npm run ng` would allow folks to be sure to use the relative angular cli version instead of the global one.

patrickmichalina reviewed (changes requested):
I'd also like to see a little bit about the `circle.yml` file. pretty straightforward pattern in there per project.

william-reynolds-flosports committed:
docs(ng-lazy-load): add better documentation, demo, and readme

william-reynolds-flosports committed:
ci(ng-lazy-load): add test job

william-reynolds-flosports committed:
docs: add ci workflow detail and use npm run for ng commands

william-reynolds-flosports committed:
docs: add missing table of contents header

william-reynolds-flosports committed:
style: fix lint errors introduced by ng cli

william-reynolds-flosports committed:
docs: add lint-fix switch for ng generate commands

william-reynolds-flosports committed:
test(ng-lazy-load): fix lint failures

We have limited control over ignoring lint problems in the demo app. This will merit further work,
because it essentially means generated files are broken when lint-fix is run.

william-reynolds-flosports committed:
test: add missing deps to tests

patrickmichalina committed:
ci

patrickmichalina committed:
readme

patrickmichalina reviewed (approved):
LGTM, thanks. I added the CI step for the lazy module to build on circle.

patrickmichalina committed:
test script

patrickmichalina committed:
memory

william-reynolds-flosports committed:
test(ng-lazy-load): add test for directive

patrickmichalina committed:
wip

patrickmichalina committed:
fix
